### PR TITLE
Fix CREATE named path projection

### LINKED ISSUE #1279: CREATE p=(n) RETURN p returns error 'p0' not defined
## When executing the query:
CREATE p=(n) RETURN p;
## FalkorDB returns the error:
'p0' not defined
## Expected Behavior:
The query should successfully create a node n, bind it to a path variable p, and return p. In Neo4j and other Cypher-compliant graph databases, this syntax is valid and should return the created node wrapped in a path.

<br/>
<hr/>

<details><summary>This repo is using Opire - what does it mean? 👇</summary><br/>💵 Everyone can add rewards for this issue commenting <code>/reward 100</code> (replace <code>100</code> with the amount).<br/>🕵️‍♂️ If someone starts working on this issue to earn the rewards, they can comment <code>/try</code> to let everyone know!<br/>🙌 And when they open the PR, they can comment <code>/claim #1279</code> either in the PR description or in a PR's comment.<br/><br/>🪙 Also, everyone can tip any user commenting <code>/tip 20 @swf2000</code> (replace <code>20</code> with the amount, and <code>@swf2000</code> with the user to tip).<br/><br/>📖 If you want to learn more, check out our <a href="https://docs.opire.dev">documentation</a>.</details>

### PR DESCRIPTION
**Pull request by nicolatrozzi**
Fixes #1279

/claim #1279

## Summary
- register named paths introduced by CREATE as path aliases after their pattern elements are validated
- extend projected named-path annotation to CREATE patterns through the same helper used for MATCH patterns
- cover CREATE named-path projection for node-only paths, relationship paths, multi-hop paths, WITH handoff, multiple named paths, and an undefined-variable guard

## Root cause
CREATE validation manually walks the path elements so newly-created node and relationship aliases do not leak into their own property expressions. That manual path skipped the named path alias itself, so a later RETURN or WITH projection of p from CREATE p=(...) was rejected as undefined before the named-path annotation pass could lower it to the existing topath expression.

## Validation
- env -u CPLUS_INCLUDE_PATH PATH=/home/majdalab/bounties/tools/peg-local/usr/bin:/home/majdalab/bounties/tools/falkordb-test-venv/bin:$PATH REDIS_SERVER=/home/majdalab/bounties/tools/redis-8.0.0/src/redis-server make flow-tests TEST=test_create_clause:testCreateClause.test04_named_path_projection SLOW=1
- env -u CPLUS_INCLUDE_PATH PATH=/home/majdalab/bounties/tools/peg-local/usr/bin:/home/majdalab/bounties/tools/falkordb-test-venv/bin:$PATH REDIS_SERVER=/home/majdalab/bounties/tools/redis-8.0.0/src/redis-server make flow-tests TEST=test_create_clause SLOW=1
- manual check: GRAPH.QUERY issue1279 "CREATE p=(n) RETURN p" returns a path and PING returns PONG

## Summary by CodeRabbit

* **New Features**
  * Named paths created in `CREATE` clauses can be referenced in subsequent projection/`RETURN` expressions, including use with `nodes()`, `relationships()`, `size()`, `length()`, and indexing into projected path elements.
* **Bug Fixes**
  * Improved `CREATE` clause scoping for named-path identifiers, and expanded duplicate-identifier detection to cover both path and edge cases.
* **Refactor**
  * Centralized named-path annotation logic for pattern/projection handling, simplifying `MATCH`, `CREATE`, and `MERGE` processing.
* **Tests**
  * Added `test04_named_path_projection` covering valid named-path projections and expected failures for redeclarations and undefined named-path references.

### CODE REVIEW

File: src/ast/ast_validations.c [hunk 1]
```diff
@@ -2475,4 +2485,3 @@ void AST_ReportErrors
 	ErrorCtx_SetError(EMSG_PARSER_ERROR, errMsg, errPos.line, errPos.column,
 			errPos.offset, errCtx, errCtxOffset);
 }
-
```
Comment (GomezGab) [inline src/ast/ast_validations.c]: avoid removing lines at the end of files.

**Review by GomezGab — COMMENTED** [review]
This looks fine. Very similar to #1716 but with the extended test suite you requested @swilly22
